src/term/mod.rs

```diff
@@ -453,13 +453,30 @@ impl<'a> Iterator for RenderableCellsIter<'a> {
                 (cell, selected)
             };
 
-            // Apply inversion and lookup RGB values
-            let mut fg_rgb = self.compute_fg_rgb(cell.fg, &cell);
-            let mut bg_rgb = self.compute_bg_rgb(cell.bg);
+            let mut invert_selection = false;
+            // Set custom selection color and selection text color, if they were provided.
+            let (bg_color, fg_color) = if selected {
+                (self.config.selection_selection_color()
+                    .unwrap_or_else(|| {
+                        // If selection color wasn't set by user we should invert cell colors.
+                        invert_selection = true;
+                        cell.bg
+                    }),
+                self.config.selection_text_color()
+                    .unwrap_or_else(|| cell.fg))
+            } else {
+                (cell.bg, cell.fg)
+            };
 
-            let bg_alpha = if selected ^ cell.inverse() {
+            // Apply inversion (if required) and lookup RGB values.
```

Comment:
This comment is a bit confusing, since the immediately following block just looks up the RGB values.

The `Apply inversions if required` comment should be added to the block which actually does the inversions (the `let bg_alpha` one).